packages/vm/src/config.rs

```diff
@@ -0,0 +1,150 @@
+use std::{collections::HashSet, path::PathBuf};
+
+use serde::{Deserialize, Serialize};
+
+use crate::Size;
+
+const DEFAULT_MEMORY_LIMIT: u32 = 512; // in pages
+/// As of March 2023, on Juno mainnet the largest value for production contracts
+/// is 485. Most are between 100 and 300.
+const DEFAULT_TABLE_SIZE_LIMIT: u32 = 2500; // entries
+
+/// We keep this number high since failing early gives less detailed error messages. Especially
+/// when a user accidentally includes wasm-bindgen, they get a bunch of unsupported imports.
+const DEFAULT_MAX_IMPORTS: usize = 100;
+
+const DEFAULT_MAX_FUNCTIONS: usize = 20_000;
+
+const DEFAULT_MAX_FUNCTION_PARAMS: usize = 100;
+
+const DEFAULT_MAX_TOTAL_FUNCTION_PARAMS: usize = 10_000;
+
+const DEFAULT_MAX_FUNCTION_RESULTS: usize = 1;
+
+/// Various configurations for the VM.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[non_exhaustive]
+pub struct Config {
+    /// Configuration for limitations placed on Wasm files.
+    /// This defines a few limits on the Wasm file that are checked during static validation before
+    /// storing the Wasm file.
+    pub wasm_limits: WasmLimits,
+
+    /// Configuration for the cache.
+    pub cache: CacheOptions,
+}
+
+impl Config {
+    pub fn new(cache: CacheOptions) -> Self {
+        Self {
+            wasm_limits: WasmLimits::default(),
+            cache,
+        }
+    }
+}
+
+/// Limits for static validation of Wasm files. These are checked before storing the Wasm file.
+/// All limits are optional because they are coming from the Go-side and have default values.
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+#[non_exhaustive]
+pub struct WasmLimits {
+    /// Maximum number of memory pages that a module can request.
+    pub memory_page_limit: Option<u32>,
```

Comment:
The naming and docs is a bit hard to understand here, esopecially since there is a different memory limit in the cache. Can we rename this to "initial memory limit (in pages)"?